backend/database.js

```diff
@@ -184,15 +185,31 @@ function throwIfFileOutsideDataDir (filename: string): string {
 }
 
 if (production || process.env.GI_PERSIST) {
+  // https://github.com/isaacs/node-lru-cache#usage
+  const cache = new LRU({
+    max: 1000
+  })
+
   sbp('sbp/selectors/overwrite', {
     // we cannot simply map this to readFile, because 'chelonia/db/getEntry'
     // calls this and expects a string, not a Buffer
     // 'chelonia/db/get': sbp('sbp/selectors/fn', 'backend/db/readFile'),
     'chelonia/db/get': async function (filename: string) {
-      const value = await sbp('backend/db/readFile', filename)
-      return Boom.isBoom(value) ? null : value.toString('utf8')
+      if (cache.has(filename)) {
+        return cache.get(filename)
+      }
```

Comment:
One can and should avoid a double-lookup by just calling `.get` and checking for `undefined` or `null` (or whatever is considered "non-existent").